src/chart_types/partition_chart/state/selectors/get_legend_items_labels.ts

```diff
@@ -24,18 +24,37 @@ import { HierarchyOfArrays, CHILDREN_KEY } from '../../layout/utils/group_by_rol
 import { Layer } from '../../specs';
 import { LegendItemLabel } from '../../../../state/selectors/get_legend_items_labels';
 import { getSettingsSpecSelector } from '../../../../state/selectors/get_settings_specs';
+import { SettingsSpec } from '../../../../specs';
 
 /** @internal */
 export const getLegendItemsLabels = createCachedSelector(
   [getPieSpecOrNull, getSettingsSpecSelector, getTree],
   (pieSpec, { legendMaxDepth }, tree): LegendItemLabel[] => {
-    if (!pieSpec || (typeof legendMaxDepth === 'number' && legendMaxDepth <= 0)) {
+    if (!pieSpec) {
       return [];
     }
-    return flatSlicesNames(pieSpec.layers, 0, tree);
+    if (isInvalidLegendMaxDepth({ legendMaxDepth })) {
+      return [];
+    }
+    const labels = flatSlicesNames(pieSpec.layers, 0, tree).filter(({ depth }) => {
+      if (typeof legendMaxDepth !== 'number') {
+        return true;
+      }
+      return depth <= legendMaxDepth;
+    });
+
+    return labels;
   },
 )(getChartIdSelector);
 
+/**
+ * Check if the legendMaxDepth from settings is not a valid number (NaN or <=0)
+ * @param legendMaxDepth - Pick<SettingsSpec, 'legendMaxDepth'>
+ */
+function isInvalidLegendMaxDepth({ legendMaxDepth }: Pick<SettingsSpec, 'legendMaxDepth'>): boolean {
```

Comment:
```suggestion
function isInvalidLegendMaxDepth(legendMaxDepth: SettingsSpec['legendMaxDepth]): boolean {
```

nit: feel free to ignore...

Reply:
done in cee4b63
thanks for the tip ❤️